SmartDeviceLink/SDLRegisterAppInterface.m

```diff
@@ -42,23 +47,40 @@ - (instancetype)initWithLifecycleConfiguration:(SDLLifecycleConfiguration *)life
                 nightColorScheme:lifecycleConfiguration.nightColorScheme];
 }
 
+- (instancetype)initWithAppName:(NSString *)appName appId:(NSString *)appId fullAppId:(nullable NSString *)fullAppId languageDesired:(SDLLanguage)languageDesired {
+    self = [self initWithAppName:appName languageDesired:languageDesired];
+    if (!self) {
+        return nil;
+    }
+
+    self.fullAppID = fullAppId.length > 0 ? fullAppId : nil;
+    self.appID = fullAppId.length > 0 ? [self.class sdlex_shortAppIdFromFullAppId:fullAppId] : appId;
```

Comment:
This should be done in the `SDLLifecycleConfiguration` in a separate private function.